retrier/retrier.go

```diff
@@ -47,16 +54,16 @@ func (r *Retrier) Run(work func() error) error {
 // returned to the caller. If the result is Retry, then Run sleeps according to the its backoff policy
 // before retrying. If the total number of retries is exceeded then the return value of the work function
 // is returned to the caller regardless.
-func (r *Retrier) RunCtx(ctx context.Context, work func(ctx context.Context) error) error {
+func (r *Retrier) RunCtx(ctx context.Context, work func(ctx context.Context, retries int) error) error {
```

Comment:
Changing the signature of the `work` function here is a breaking change for other users of this method. If you really need the retry count in your work function, you will need to add another `Run*` method that chains the call (in the same way that `Run` chains to `RunCtx`).